compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -491,20 +518,71 @@ LogicalResult insertLogicalObjectFifoAccess(ModuleOp moduleOp) {
             auto accessOp = rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
                 rewriter.getUnknownLoc(), std::get<0>(value),
                 std::get<1>(value));
-            linalgOp->setOperand(idx, accessOp);
-          } else if (auto type =
-                         llvm::dyn_cast<MemRefType>(operand.get().getType())) {
-            Value memref = operand.get();
+            op->setOperand(idx, accessOp);
+          } else if (isa<memref::AllocOp, memref::SubViewOp,
+                         vector::TransferReadOp, arith::ExtFOp>(
+                         operand.get().getDefiningOp())) {
+            OpOperand *opOperand = &operand;
+            Operation *defOpOfOperand = operand.get().getDefiningOp();
+            assert(defOpOfOperand && "expected a defining op of the operand");
+            while (defOpOfOperand && !isa<memref::AllocOp>(defOpOfOperand)) {
+              opOperand = &(defOpOfOperand->getOpOperand(0));
+              defOpOfOperand = opOperand->get().getDefiningOp();
+            }
+            Value memref = opOperand->get();
+            MemRefType type = dyn_cast<MemRefType>(memref.getType());
+            assert(type && "expected memref type");
             rewriter.setInsertionPoint(coreOp);
             auto logicalObjectFifo =
                 rewriter.create<AMDAIE::LogicalObjectFifoFromMemrefOp>(
                     rewriter.getUnknownLoc(), LogicalObjectFifoType::get(type),
                     memref);
             rewriter.setInsertionPointToStart(coreOp.getBody());
-            auto accessOp = rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
-                rewriter.getUnknownLoc(), logicalObjectFifo,
-                AMDAIE::MemoryAccess::None);
-            linalgOp->setOperand(idx, accessOp);
+
+            AMDAIE::LogicalObjectFifoAccessOp accessOp;
+            if (memrefToLogicalObjectFifo.contains(memref)) {
+              std::tuple<AMDAIE::LogicalObjectFifoFromMemrefOp,
+                         AMDAIE::MemoryAccess>
+                  value = memrefToLogicalObjectFifo[memref];
+              accessOp = rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
+                  rewriter.getUnknownLoc(), std::get<0>(value),
+                  std::get<1>(value));
+            } else {
+              accessOp = rewriter.create<AMDAIE::LogicalObjectFifoAccessOp>(
+                  rewriter.getUnknownLoc(), logicalObjectFifo,
+                  AMDAIE::MemoryAccess::None);
+            }
+            newMemrefToLogicalObjectFifoAccess[memref] = accessOp;
+            opOperand->getOwner()->setOperand(opOperand->getOperandNumber(),
+                                              accessOp);
+          }
+        }
+
+        // Handle vector.transfer_write.
```

Comment:
Could you clarify in the comment what's needed to handle `vector.transfer_write` specifically?